docs/design.md

```diff
@@ -82,28 +81,146 @@ By splitting your code like this, you can easily keep all your systems and resou
 A good rule of thumb is to have one plugin per file,
 but feel free to leave out a plugin if your file does not need to do anything with the `App`.
 
-## Widgets
+## Screen States
+
+### Pattern
+
+Use the [`Screen`](../src/screen/mod.rs) enum to represent your game's screens as states:
+
+```rust
+#[derive(States, Debug, Hash, PartialEq, Eq, Clone, Default)]
+pub enum Screen {
+    #[default]
+    Splash,
+    Loading,
+    Title,
+    Credits,
+    Gameplay,
+    Victory,
+    Leaderboard,
+    MultiplayerLobby,
+}
+```
+
+For each screen, create a plugin that handles the setup and teardown in the
+[`OnEnter`](https://docs.rs/bevy/latest/bevy/prelude/struct.OnEnter.html) and
+[`OnExit`](https://docs.rs/bevy/latest/bevy/prelude/struct.OnExit.html) schedules.
+You can also mark an entity to despawn on exit by giving it the
```

Comment:
Sorry to continue nitpicking, but can we be a bit more firm here? Instead of "can", you "should despawn entities belonging to this screen by…" or something like that